snuba/web/views.py

```diff
@@ -134,9 +136,18 @@ def check_clickhouse(
         entities = itertools.chain(
             *[dataset.get_all_entities() for dataset in datasets]
         )
-        storages = list(
-            itertools.chain(*[entity.get_all_storages() for entity in entities])
-        )
+
```

Comment:
I would recommend moving the logic of filtering which storage/cluster needs to be checked(including the experimental and disabled_datasets) into a method of its own. The `check_connection` method is doing too much right now and has too much context baked into it. Moving it out would help code readability and help write tests for that logic.

Reply:
Agreed. Moved into `filter_checked_storages` function.